feature: Base CSV Checking and RVM Addition

sfrinaldi opened the pull request:
Tested on develop branch to verify the added attribute ties done in this branch for the RVM don't break on ones that do not have this information / csv headers yet. Appears to work still without breaking.

Feature includes doorstop base check during the `doorstop import` command for csv files. Workflow will still continue but a warning_log.txt file will generate (as well as console warning messages during the run) with just the duplicate UID errors it found when parsing through each csv file. 

Also skips over / ignores comment lines now for the invalid UID warnings that would generate before. 

Adding RVM attributes for being able to pull those values easier from doorstop (in the correct manner) for implementing with the modified rtv_doorstop fork.  Removed testing files so repo doesn't need to be removed after cloning (or to be implemented as a submodule or something else).  (Testing files of requirements cause there to be multiple 'roots' detected as there is a built tree already)

**Example warning_log.txt:** 

![image](https://github.com/user-attachments/assets/cc82f45a-e38f-44a2-8756-69b6b9d82356)

More features will be added to this whether in a separate repo (requirements-tools) or using available tools (like [csv-lint](https://github.com/Clever/csvlint)). Pros for merging this initial feature earlier is current requirements being developed / reviewed in Gitlab will be able to get this warning_log.txt (locally / gitlab workflow will need to be adjusted for adding this file as an artifact) and duplicate uid messages in the workflow logs.

sfrinaldi committed:
Initial Adjustments / Not completed

Testing out having RVM built into doorstop / seeing how it would be for automation. Not completed yet / first commits. This is non-functional at the moment.

sfrinaldi committed:
Update markdown.py

Commenting out RVM

sfrinaldi committed:
Update item.py

Adjusting new RTM values.

sfrinaldi committed:
Updating for rtm attributes

sfrinaldi committed:
Removing test reqs

Will allow repo to be in pearl_requirements / shorten time (could be a submodule).

sfrinaldi committed:
Changing loging level

Trying to just have logging write to a file for ones that are warnings.

sfrinaldi committed:
Adding csvcimple

sfrinaldi committed:
Update importer.py

sfrinaldi committed:
Update importer.py

sfrinaldi commented:
@patrickingraham Let me know if you think this is worth pulling to main in its initial state so that the current branches (especially when compiling locally) can see this information for when requirements are being adjusted. Otherwise I can just wait on this PR / continue to work on it.

douglase commented:
@sfrinaldi have you tested against https://github.com/douglase/doorstop_requirements_template/ or do we need to update that too?

sfrinaldi commented:
Will double check!

sfrinaldi commented:
Alright, so I checked it out and for that repo it looks like its what is used in [requirements_tools](https://github.com/uasal/requirements_tools) but also a lot of the functionality for the newer doorstop (as it can publish to latex / markdown / excel actually now and not just html) doesn't require pandocs. I see there is a UASAL fork [uasal/doorstop_requirements_template](https://github.com/uasal/doorstop_requirements_template) for that repo you linked ([douglase/doorstop-requirements-template](https://github.com/douglase/doorstop_requirements_template/)) which looks like it is out of date in comparison. 

The graphviz is still being generated and added (in pretty much the same fashion but we have it stop at a certain level (4) as then it becomes a bit unreadable). Did we want to have it integrated / use travisci? Let me know if there was something else you were referring too that I didn't cover.

sfrinaldi committed:
Adding no include_graphics option

Edits to latex.py to allow for latex template to not have include_graphics without erroring out.

sfrinaldi committed:
Fix to graphic check in template_data

For latex.py, fixing wrapper statement and check for if 'include_graphics' key is present in template_data.

sfrinaldi committed:
Removing default doctitle

Having doctitle be able to be set in the template for latex publishing.

sfrinaldi committed:
Adding matrix template setting ability for latex

Will check template_data for if the traceability

sfrinaldi committed:
Removing tutorial test files

Prevents the need of doorstop needing to be re-cloned and remove for each run.

sfrinaldi committed:
Update latex.py

Spelling correction.

sfrinaldi committed:
Adding traceability_matrix to default template

Adding the option to the default template for including the traceability_matrix to the published latex documentation.

sfrinaldi committed:
Template Adjustments

sfrinaldi committed:
Update .gitignore

patrickingraham commented on doorstop/core/importer.py:
```diff
@@ -249,7 +256,15 @@ def _itemize(header, data, document, mapping=None):
             # Convert values for particular keys
             if key in ("uid", "id"):  # 'id' for backwards compatibility
-                uid = value
+                if value is not False:
```
if True

sfrinaldi replied:
Pushed changes!

sfrinaldi replied:
Just realized an interesting issue with this. Reverted the changes for this the way this is working right now I believe from line 154 change in this document `importer.py`. Will circle back to this for fixes / elif is needed at the moment the way its structured for the log output / warnings to skip over comments and empty lines.

patrickingraham replied:
hmm, probably has something to do with the `is` -- meaning that value changes types... not ideal, but ok.

sfrinaldi committed:
Applied edit per #8

Changed not False to is True.

patrickingraham commented on doorstop/cli/tests/files/exported.tsv:
this is shown as having all the content removed but not the actual file. is this correct?
There are several like this.

sfrinaldi replied:
Fixed / show as removed now.

sfrinaldi committed:
Removing test file defaults

sfrinaldi committed:
Revert "Applied edit per #8"

This reverts commit eac485a5e39edff4fa8be6ba7f1454872153d425.

sfrinaldi committed:
u/sfrinaldi/#1-rvm-edits (#15)

* Edits to RVM items in item.py

Correcting comments in loading/setting methods as well.

* Add default sep value to document.py

* Change reorder to false in tree.py

* Revert "Add default sep value to document.py"

This reverts commit 1cb26f176b04431734ac69b4d4f4bcbdc50b3d57.

* Revert "Change reorder to false in tree.py"

This reverts commit f251b936f568aee7205dcbdf081ffc5145ba6e72.

* Add template_data[rvm] to latex.py

* Add rvm to doorstop.yml template

* Add rvm to wrapper in latex.py

* Add booktabs package to latex template

* Add \landscape for RVM section to latex.py

* Add lscape package to latex templates

* Add rotating/adjustbox packages to latex templates

* Edit wrapper for RVM in latex.py

* Testing sidewaystable in latex.py wrapper

* Test to markdown for latex

* Edit RVM wrapper in latex.py

* Update latex templates

* Edits to packages in latex templates

* Remove test markdown setup in doorstop.cls

* Fix to rvm wrapper in latex.py

* Add \sloppy to doorstop.yml

* Package update to doorstop.cls

* Fix to rvm wrapper in latex.py

sfrinaldi commented:
Renamed to better reflect what this PR is now.

sfrinaldi commented on doorstop/core/files/templates/latex/doorstop.yml:
```diff
@@ -34,7 +41,8 @@ usepackage:
 before_begin_document:
   # Define the title for the published document output
-  - \def\docname{Pearl Requirements}
+  - \def\doctitle{Pearl Requirements}
```
Thats in the latex template at least but was thinking we might want to move it to the config.yml file or something like that? You can have multiple templates (just need to specify which in the command if there are). 

I'm thinking we have the templates move and get pulled within the requirements repos in general (pretty much like how its doing for the WCC local deploy actually). I didn't push that out just yet but it already had that functionality. I can change the 'base' doorstop template to something else like 'Requirements'.

sfrinaldi replied:
I'll open an issue for this discussion though for now!

sfrinaldi commented:
Doing a quick check / test on pearl_requirements with the updated branches for doorstop-ewan and the related rtm_doorstop branch after a PR has gone through. (Last check for any issues before final merging for this PR).
